src/index.ts

```diff
@@ -668,11 +668,22 @@ async function prepareTxs({
 	passwords = [],
 	provider,
 }: interfaces.IPrepareTxsParams): Promise<interfaces.IPrepareTxsResponse> {
+	if (!provider) {
+		try {
+			provider = await getDefaultProvider(String(linkDetails.chainId))
+		} catch (error) {
+			throw new interfaces.SDKStatus(
+				interfaces.EPrepareCreateTxsStatusCodes.ERROR_GETTING_DEFAULT_PROVIDER,
+				'Error getting the default provider'
+			)
+		}
+	}
+
```

Comment:
I don't think a try catch block is required here - getDefaultProvider already throws if not successful:

```
} catch (error) {
		throw new Error('No alive provider found for chainId ' + chainId)
	}
	````